### PR TITLE
 Refresh the Console View Step Header Cards

### PR DESCRIPTION
**Pull request by stuartrowe**
Fixes https://github.com/jenkinsci/pipeline-graph-view-plugin/issues/346
- Included status icon on step header
- Faded the color of the step header background so the status icon is visible
- Fixed the color and spacing of the right-hand fullscreen, link and expand icons
- Changed the step header text color for consistency with the tree view
- Fixed a bug where splitting the step displayName on the last '-' would split a valid step label

**Before**
image
image

**After**
image
image

### Testing done

```[tasklist]
### Submitter checklist
- [X] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [X] Ensure that the pull request title represents the desired changelog entry
- [X] Please describe what you did
- [X] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

**Comment by timbrown5** [comment]
I agree this is more in keeping with the rest of the view, but worry it makes it harder to quickly scan for where there are errors.
You could take a cue from the Blue Ocean view and set the background of the first div: 
image

**Comment by stuartrowe** [comment]
@timbrown5 are you suggesting to color the background behind the status icons I've added? That would directly copy Blue Ocean, but it doesn't fit with the current look of Pipeline Graph View. Also, would it be difficult to create white/negative colored status icons to show up against that background color?
image

**Comment by timbrown5** [comment]
You could fade the colour to make the icon more visible.
I guess my general worry is that making it more in keeping could make it
less useful.

File: src/main/frontend/pipeline-console-view/pipeline-console/main/pipeline-console.scss [hunk 1]
```diff
@@ -17,14 +17,26 @@
...
+  --step-bg-aborted: color-mix(in srgb, var(--purple) 50%, black);
+  --step-bg-paused: color-mix(in srgb, var(--blue) 50%, black);
 }
 
 @media (prefers-color-scheme: dark) {
   [data-theme="dark-system"],
   [data-theme="dark-system"] {
     --card-background: hsl(230deg 14% 23%);
-    --step-text-color: hsl(230deg 14% 23%);
+    //--step-text-color: hsl(230deg 14% 23%);
```
Comment (timja) [inline src/main/frontend/pipeline-console-view/pipeline-console/main/pipeline-console.scss:33]: is this still needed?

File: src/main/frontend/pipeline-console-view/pipeline-console/main/pipeline-console.scss [hunk 2]
```diff
@@ -17,14 +17,26 @@
 
 [data-theme="dark"] {
   --card-background: hsl(230deg 14% 23%);
-  --step-text-color: hsl(230deg 14% 23%);
+  //--step-text-color: hsl(230deg 14% 23%);
```
Comment (timja) [inline src/main/frontend/pipeline-console-view/pipeline-console/main/pipeline-console.scss:20]: is this still needed?

**Review by timja — COMMENTED** [review]
LGTM, tested locally with a variety of setups, anything left on this?

**Comment by stuartrowe** [comment]
> LGTM, tested locally with a variety of setups, anything left on this?

I want to revisit splitting the step name and title. Currently it's combined on a single line and we lose the title if the name is long enough.

I'll also get some screenshots together to update the MR description with the latest changes.

**Comment by stuartrowe** [comment]
The last issue I have is the layout of the status icon, name and title on the console modal. I haven't been able to recreate the layout used by the step header where the status icon is to the left of the display name and then on the next line the title is included aligned with the display name. Maybe someone more experience with the mui reach components can take this on?

Currently the step display and title are on the same line.
image

**Comment by timja** [comment]
I don't understand what you mean, maybe if you mocked it up by adjusting the HTML or drawing it?

happy to merge this as-is though

**Comment by stuartrowe** [comment]
Here's what I want moved on the modal:
console_modal_before
to
console_modal_after

**Comment by timja** [comment]
Doesn’t that end up taking a lot more vertical space? What would be the benefit in moving this?

**Comment by stuartrowe** [comment]
It would be consistent with the current layout. Although the user would have seen the console log card already in order to navigate to the modal. They won't be interested in the step name/title as much as the log. The loss of vertical space doesn't seem worth it. I'll leave it as is as you suggested.

**Pull request merged**